### DISCUSSION TITLE
[Question][Hw2] Order of a model of satisfiable case

### DISCUSSION THREAD

**Discussion by KihongHeo** [prosyslab-classroom/cs424-program-reasoning#371] [Q&A]
Hello. This is a duplicated question. Copyright: #51.

I have been implementing the solvers for Hw2.
I believe I have correctly implemented the program equivalence checker, in that I think it prints "satisfiable" along with a model when the programs are not equivalent, and "unsatisfiable" when they are equivalent.

[Small change]

However, for some reason, when printing a model in which the programs give different results, my implementation prints the variables in a different order than was provided on Github.

[Small change]

Below is a picture of my testing with the eq3.scm file.
Instead of printing in tgt_x -> src_x -> src_y order,
it does in tgt_x -> src_y -> src_x order.

My question is, does the order of printing a satisfying model matter?

Thank you.

===
Sorry for unnecessary notification emails.

**Comment by KihongHeo** [comment]
I am surprised that there is no good duplicated issue detector in [Github Market Place](https://github.com/marketplace/actions/). The [existing one](https://github.com/marketplace/actions/potential-duplicates) is not very satisfactory.

If some of you are interested in developing such a Github bot with me, let me know.

**Comment by Allgot** [comment]
Well, I'm absolutely interested in it but I'm afraid I don't know which ability is required. Could you explain it in detail?

**Comment by KihongHeo** [comment]
I don't know, too. 
But if we really want to do something, we will know and we must know! Don't worry.

Maybe 
- programming in JavaScript to make a [custom action](https://docs.github.com/en/actions/creating-actions/about-custom-actions)
- basic skills for docker container
- basic programming skills for REST APIs to use a [text similarity check](https://www.twinword.com/api/text-similarity.php) and Github APIs

If you are interested in, email me (kihong.heo@kaist.ac.kr).
